birdyboard.py

```diff
@@ -0,0 +1,443 @@
+import pickle
+
+
+class Birdyboard:
+
+    def __init__(self):
+        self.users = []
+        self.user_name = None
+
+        self.chirps_library = {"public": [], "private": []}
+        # these variables are assigned when a specific chirp thread is being viewed or added to.
+        self.public_or_private = ""
+        self.chirp_index = None
+
+# ############################################
+# ######## UNLOGGED IN TOP LEVEL MENU ########
+# ############################################
+
+    def unlogged_in_menu_print(self):
+        """
+        prints the choices for an unlogged-in user, at start of app.
+        Arguments: none
+        """
+        print("Welcome. You are not logged in.\n")
+        print("1. new user\n2. log in\n3. view public chirps\n'x' to exit.\n")
+
+    def unlogged_in_menu_next_step(self):
+        """
+        requests input for top level menu functionality while the user is not logged in. Currently does not do anything for user login. secret fourth option to add a chirp but later that will be only an option if you're logged in.
+        Arguments: none
+        """
+        next_step = input(">> ")
+        if next_step == "1":  # create a user.
+            print("creating a new user.")
+            self.create_a_user_menu()
+        elif next_step == "2":  # log in to a current user.
+            self.view_users_menu()
+            self.users_menu_next_step()
+        elif next_step == "3":  # view public chirps. Will not have the option of commenting.
+            self.deserialize_chirps_library()
+            self.public_or_private = "public"
+            self.view_chirps()
+            self.view_chirps_next_step()
+        elif next_step == "x":  # exit.
+            print("goodbye.")
+            exit()
+        else:
+            print("command not found.")
+            self.unlogged_in_menu_next_step()
+
+# ##########################################
+# ######## LOGGED IN TOP LEVEL MENU ########
+# ##########################################
+
+    def logged_in_menu_print(self):
+        print("welcome " + self.user_name + "!")
+        print("1. log out\n2. view public chirps\n3. view private chirps\n4. New public chirp thread\n5. new private chirp thread\n'x' to exit.\n")
+
+    def logged_in_menu_next_step(self):
```

Comment:
the reason I'm running this way is because of the way I want the menus to
print. I don't want to print all the menu options every time someone types
in the wrong thing, so I pulled them out. I'll put them into "prints"
alone, but I'm not going to put them into the next steps functions.

On Tue, Aug 2, 2016 at 2:15 PM, Ryan Tanay <notifications@github.com> wrote:

> In birdyboard.py
> <https://github.com/nss-day-cohort-13/birdyboard-MeganCFord/pull/3#discussion_r73219119>
> :
>
> > +        elif next_step == "x":  # exit.
> > +            print("goodbye.")
> > +            exit()
> > +        else:
> > +            print("command not found.")
> > +            self.unlogged_in_menu_next_step()
> > +
> > +# ##########################################
> > +# ######## LOGGED IN TOP LEVEL MENU ########
> > +# ##########################################
> > +
> > +    def logged_in_menu_print(self):
> > +        print("welcome " + self.user_name + "!")
> > +        print("1. log out\n2. view public chirps\n3. view private chirps\n4. New public chirp thread\n5. new private chirp thread\n'x' to exit.\n")
> > +
> > +    def logged_in_menu_next_step(self):
>
> Same as above here, you don't really need a whole method just to print 2
> lines, the above method could be rolld into this one.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/nss-day-cohort-13/birdyboard-MeganCFord/pull/3/files/c29e8f0a6183667cfe0e6b20fd4ee0b018dec442#r73219119>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/ARUAVWhD9ylZYhIDNEL5CwhttZB360KXks5qb5dugaJpZM4JaAC3>
> .
>